src/modules/admin/admin.controller.js

```diff
@@ -0,0 +1,70 @@
+const Admin = require("./admin.model.js");
+
+// check admin exsists or not for given email
+const checkAdminExsists = async (email) => {
+  try {
+    const adminExsists = await Admin.findOne({ where: { email: email } });
+    return adminExsists;
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+// function for register admin in system.
+const registerAdmin = async (req, res) => {
+  const data = { ...req.body };
+  const checkAdmin = await checkAdminExsists(data.email);
+  if (!checkAdmin) {
+    try {
+      const registerAdmin = await Admin.create(data);
+      res.send({
+        message:
+          "Admin registration successful ! Please check email for active account.",
+        data: registerAdmin.dataValues,
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  } else {
+    res.send({ error: "An account already exsists with this email." });
+  }
+};
+
+// function for singin in admin dashboard
+const adminSignIn = async (req, res) => {
+  const data = { ...req.body };
+  const response = await checkAdminExsists(data.email);
+  if (response) {
+    try {
+      data.email == response.dataValues.email &&
```

Comment:
why we check the email twice. we already check it in line 37 & 37

Reply:
Vai please check , I did some update. Is it ok?